File: bmpm-php/BMPMGeneric.php
<?php

class BMPMGeneric extends BMPM {
	function __construct() {
		$this->setType(BMPM::TYPE_GENERIC);
		
		include "gen/approxcommon.php";
		$this->setApproxCommon($approxCommon);
		
		include "gen/exactcommon.php";
		$this->setExactCommon($exactCommon);

		include "gen/languagenames.php";
		$this->setLanguageNames($languages);
		
		include "gen/lang.php";
		$this->setAllLanguagesBitmap($all);
		$this->setLanguageRules($languageRules);

		$rules = array();
		include "gen/rulesany.php";
		include "gen/rulesarabic.php";
		include "gen/rulescyrillic.php";
		include "gen/rulesczech.php";
		include "gen/rulesdutch.php";
		include "gen/rulesenglish.php";
		include "gen/rulesfrench.php";
		include "gen/rulesgerman.php";
		include "gen/rulesgreek.php";
		include "gen/rulesgreeklatin.php";
		include "gen/ruleshebrew.php";
		include "gen/ruleshungarian.php";
		include "gen/rulesitalian.php";
		include "gen/ruleslatvian.php";
		include "gen/rulespolish.php";
		include "gen/rulesportuguese.php";
		include "gen/rulesromanian.php";
		include "gen/rulesrussian.php";
		include "gen/rulesspanish.php";
		include "gen/rulesturkish.php";
		$this->setRules($rules);

		$approx = array();
		include "gen/approxany.php";
		include "gen/approxarabic.php";
		include "gen/approxcyrillic.php";
		include "gen/approxczech.php";
		include "gen/approxdutch.php";
		include "gen/approxenglish.php";
		include "gen/approxfrench.php";
		include "gen/approxgerman.php";
		include "gen/approxgreek.php";
		include "gen/approxgreeklatin.php";
		include "gen/approxhebrew.php";
		include "gen/approxhungarian.php";
		include "gen/approxitalian.php";
		include "gen/approxlatvian.php";
		include "gen/approxpolish.php";
		include "gen/approxportuguese.php";
		include "gen/approxromanian.php";
		include "gen/approxrussian.php";
		include "gen/approxturkish.php";
		$this->setApprox($approx);

		$exact = array();
		include "gen/exactany.php";
		include "gen/exactarabic.php";
		include "gen/exactcyrillic.php";
		include "gen/exactczech.php";
		include "gen/exactdutch.php";
		include "gen/exactenglish.php";
		include "gen/exactfrench.php";
		include "gen/exactgerman.php";
		include "gen/exactgreek.php";
		include "gen/exactgreeklatin.php";
		include "gen/exacthebrew.php";
		include "gen/exacthungarian.php";
		include "gen/exactitalian.php";
		include "gen/exactlatvian.php";
		include "gen/exactpolish.php";
		include "gen/exactportuguese.php";
		include "gen/exactromanian.php";
		include "gen/exactrussian.php";
		include "gen/exactturkish.php";
		$this->setExact($exact);


		return $this;
	}
}

?>

File: bmpm-cpp/BMPMGeneric.cpp
#include "cpp/php2cpp.h"


#ifndef BMPMGENERIC_CPP
#define BMPMGENERIC_CPP

#include "BMPMGeneric.h"

BMPMGeneric::BMPMGeneric()
    {
        this->setType(BMPM::TYPE_GENERIC);
        #include "gen/approxcommon.h"
        this->setApproxCommon(approxCommon);
        #include "gen/exactcommon.h"
        this->setExactCommon(exactCommon);
        #include "gen/languagenames.h"
        this->setLanguageNames(languages);
        #include "gen/lang.h"
        this->setAllLanguagesBitmap(all);
        this->setLanguageRules(languageRules);
        rules = {};
        #include "gen/rulesany.h"
        #include "gen/rulesarabic.h"
        #include "gen/rulescyrillic.h"
        #include "gen/rulesczech.h"
        #include "gen/rulesdutch.h"
        #include "gen/rulesenglish.h"
        #include "gen/rulesfrench.h"
        #include "gen/rulesgerman.h"
        #include "gen/rulesgreek.h"
        #include "gen/rulesgreeklatin.h"
        #include "gen/ruleshebrew.h"
        #include "gen/ruleshungarian.h"
        #include "gen/rulesitalian.h"
        #include "gen/ruleslatvian.h"
        #include "gen/rulespolish.h"
        #include "gen/rulesportuguese.h"
        #include "gen/rulesromanian.h"
        #include "gen/rulesrussian.h"
        #include "gen/rulesspanish.h"
        #include "gen/rulesturkish.h"
        this->setRules(rules);
        exact = {};
        #include "gen/exactany.h"
        #include "gen/exactarabic.h"
        #include "gen/exactcyrillic.h"
        #include "gen/exactczech.h"
        #include "gen/exactdutch.h"
        #include "gen/exactenglish.h"
        #include "gen/exactfrench.h"
        #include "gen/exactgerman.h"
        #include "gen/exactgreek.h"
        #include "gen/exactgreeklatin.h"
        #include "gen/exacthebrew.h"
        #include "gen/exacthungarian.h"
        #include "gen/exactitalian.h"
        #include "gen/exactlatvian.h"
        #include "gen/exactpolish.h"
        #include "gen/exactportuguese.h"
        #include "gen/exactromanian.h"
        #include "gen/exactrussian.h"
        #include "gen/exactspanish.h"
        #include "gen/exactturkish.h"
        this->setExact(exact);
        approx = {};
        #include "gen/approxany.h"
        #include "gen/approxarabic.h"
        #include "gen/approxcyrillic.h"
        #include "gen/approxczech.h"
        #include "gen/approxdutch.h"
        #include "gen/approxenglish.h"
        #include "gen/approxfrench.h"
        #include "gen/approxgerman.h"
        #include "gen/approxgreek.h"
        #include "gen/approxgreeklatin.h"
        #include "gen/approxhebrew.h"
        #include "gen/approxhungarian.h"
        #include "gen/approxitalian.h"
        #include "gen/approxlatvian.h"
        #include "gen/approxpolish.h"
        #include "gen/approxportuguese.h"
        #include "gen/approxromanian.h"
        #include "gen/approxrussian.h"
        #include "gen/approxspanish.h"
        #include "gen/approxturkish.h"
        this->setApprox(approx);
        return this;
    }

#else
#endif